io/io/src/TBufferJSON.cxx

```diff
@@ -3618,20 +4358,19 @@ Int_t TBufferJSON::WriteObjectAny(const void *obj, const TClass *ptrClass, Bool_
 Int_t TBufferJSON::WriteClassBuffer(const TClass *cl, void *pointer)
 {
 
-   //build the StreamerInfo if first time for the class
+   // build the StreamerInfo if first time for the class
    TStreamerInfo *sinfo = (TStreamerInfo *)const_cast<TClass *>(cl)->GetCurrentStreamerInfo();
-   if (sinfo == 0) {
-      //Have to be sure between the check and the taking of the lock if the current streamer has changed
+   if (!sinfo) {
+      // Have to be sure between the check and the taking of the lock if the current streamer has changed
       R__LOCKGUARD(gInterpreterMutex);
       sinfo = (TStreamerInfo *)const_cast<TClass *>(cl)->GetCurrentStreamerInfo();
-      if (sinfo == 0) {
+      if (!sinfo) {
          const_cast<TClass *>(cl)->BuildRealData(pointer);
          sinfo = new TStreamerInfo(const_cast<TClass *>(cl));
          const_cast<TClass *>(cl)->SetCurrentStreamerInfo(sinfo);
          const_cast<TClass *>(cl)->RegisterStreamerInfo(sinfo);
          if (gDebug > 0)
-            printf("Creating StreamerInfo for class: %s, version: %d\n",
-                   cl->GetName(), cl->GetClassVersion());
+            printf("Creating StreamerInfo for class: %s, version: %d\n", cl->GetName(), cl->GetClassVersion());
```

Comment:
Never use raw std::cout, printf, etc.
Always use the ROOT printing routine so that the user can filter and/or redirect the output. (Yes, I realize that a pre-existing line :) ).

Reply:
Code was copied as is from TBufferFile:

https://github.com/root-project/root/blob/master/io/io/src/TBufferFile.cxx#L3980

Can it be, that printf used because of mutex locking?

Reply:
I don't think so.  What you copied is just ancient, incorrect code :)

Reply:
>  What you copied is just ancient, incorrect code :)

Should we change it in TBufferFile as well?

Reply:
Yes (technically we should change all (most?) use of printf through core/io/tree (but this is of course beyond the scope of this PR :) )